panther_analysis_tool/main.py

```diff
@@ -888,28 +908,54 @@ def run_tests(
         # check expected result
         if result != unit_test["ExpectedResult"]:
             test_result["outcome"] = "FAIL"
+            failed_tests[analysis.get("PolicyID") or analysis["RuleID"]].append(unit_test["Name"])
+
+        # print results
+        print("\t[{}] {}".format(test_result["outcome"], unit_test["Name"]))
 
-        # check dedup and title function return non-None
+        # check custom function return non-None
         # Only applies to rules which match an incoming event
         if unit_test["ExpectedResult"]:
-            for func in ["dedup", "title"]:
+            for func in CUSTOM_FUNCTIONS:
                 if analysis_funcs.get(func):
-                    if not analysis_funcs[func](test_case):
+                    try:
+                        if mock_methods:
+                            with patch.multiple(analysis_funcs["module"], **mock_methods):
+                                func_out = analysis_funcs[func](test_case)
+                                if not func_out:
+                                    test_result[func] = "FAIL"
+                                    test_result["outcome"] = "FAIL"
+                        else:
+                            func_out = analysis_funcs[func](test_case)
+                            if not func_out:
+                                test_result[func] = "FAIL"
+                                test_result["outcome"] = "FAIL"
+                    except Exception as err:  # pylint: disable=broad-except
+                        func_out = err
                         test_result[func] = "FAIL"
                         test_result["outcome"] = "FAIL"
-
-        if test_result["outcome"] == "FAIL":
-            failed_tests[analysis.get("PolicyID") or analysis["RuleID"]].append(unit_test["Name"])
-
-        # print results
-        print("\t[{}] {}".format(test_result["outcome"], unit_test["Name"]))
-        for func in ["dedup", "title"]:
-            if analysis_funcs.get(func) and unit_test["ExpectedResult"]:
-                print(
-                    "\t\t[{}] [{}] {}".format(
-                        test_result[func], func, analysis_funcs[func](test_case)
+                    func_out_valid_type = False
+                    # severity value validation
+                    if func == "severity":
+                        if str(func_out).upper() not in VALID_SEVERITIES:
+                            func_out = AssertionError(
+                                f'Expected severity to be any of the following: '
+                                f'{str(VALID_SEVERITIES)}, got [{str(func_out)}] instead.'
+                            )
+                            test_result[func] = "FAIL"
+                            test_result["outcome"] = "FAIL"
```

Comment:
Not sure if this is better or worse, but I think you can do this 
```suggestion
                            test_result[func] = test_result["outcome"] = "FAIL"
```
to save a few lines here and everywhere this is repeated.